src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsicsAnalyzer.cs

```diff
@@ -0,0 +1,353 @@
+            {
+                HashSet<IMethodSymbol> methodSymbols = methodSymbolSets[(int)ruleKind];
+
+                IEnumerable<IMethodSymbol> members =
+                    typeSymbol.GetMembers(name)
+                              .OfType<IMethodSymbol>()
+                              .Where((m) => m.Parameters.Length == 2 &&
+                                            m.ReturnType is INamedTypeSymbol namedReturnTypeSymbol &&
+                                            namedReturnTypeSymbol.Arity == 1 &&
+                                            ((supportedTypes.Length == 0) || supportedTypes.Contains(namedReturnTypeSymbol.TypeArguments[0].SpecialType)) &&
+                                            (m.Parameters[1].Type.SpecialType is SpecialType.System_Byte or SpecialType.System_Int32) &&
+                                            SymbolEqualityComparer.Default.Equals(m.Parameters[0].Type, namedReturnTypeSymbol));
+
+                methodSymbols.AddRange(members);
+            }
+
+            static void AddUnaryOperatorMethods(ImmutableArray<HashSet<IMethodSymbol>> methodSymbolSets, string name, INamedTypeSymbol typeSymbol, RuleKind ruleKind, params SpecialType[] supportedTypes)
+            {
+                HashSet<IMethodSymbol> methodSymbols = methodSymbolSets[(int)ruleKind];
+
+                IEnumerable<IMethodSymbol> members =
+                    typeSymbol.GetMembers(name)
+                              .OfType<IMethodSymbol>()
+                              .Where((m) => m.Parameters.Length == 1 &&
+                                            m.ReturnType is INamedTypeSymbol namedReturnTypeSymbol &&
+                                            namedReturnTypeSymbol.Arity == 1 &&
+                                            ((supportedTypes.Length == 0) || supportedTypes.Contains(namedReturnTypeSymbol.TypeArguments[0].SpecialType)) &&
+                                            SymbolEqualityComparer.Default.Equals(m.Parameters[0].Type, namedReturnTypeSymbol));
+
+                methodSymbols.AddRange(members);
+            }
+        }
+
+        private void AnalyzeInvocation(OperationAnalysisContext context, ImmutableArray<HashSet<IMethodSymbol>> methodSymbolSets)
+        {
+            if (context.Operation is not IInvocationOperation invocation)
+            {
+                return;
+            }
+
+            IMethodSymbol targetMethod = invocation.TargetMethod;
+
+            for (int i = 0; i < methodSymbolSets.Length; i++)
+            {
+                HashSet<IMethodSymbol> methodSymbols = methodSymbolSets[i];
+
+                if (!methodSymbols.Contains(targetMethod, SymbolEqualityComparer.Default))
+                {
+                    continue;
+                }
+
+                if (IsSupported(invocation, (RuleKind)i))
```

Comment:
this will matter later, for a commetn i place in the C# subclass, but you can't assume taht jsut because a symbol bound, that the syntax is of a particular shape.  For example, a call to a method may be missing arguments (or have too many arguments) and may still bind (thanks to error recovery tolerance).